webtau-cli/src/test/java/com/twosigma/webtau/cli/CliJavaTest.java

```diff
@@ -51,7 +56,53 @@ public void envVars() {
         nixOnly(() -> {
             cli.run("scripts/hello", cli.env("NAME", "Java"), (exitCode, output, error) -> {
                 output.should(contain("hello world Java"));
+                error.should(contain("error line two"));
             });
         });
     }
+
+    @Test
+    public void docCapture() {
+        nixOnly(() -> {
+            cli.run("scripts/hello", ((output, error) -> {
+                output.should(contain("line in the middle"));
+                output.should(contain(Pattern.compile("line in the")));
+                output.should(contain("more text"));
+
+                error.should(contain("error line one"));
+            }));
+
+            String artifactName = "hello-script";
+            cli.doc.capture(artifactName);
+
+            validateCapturedDocs(artifactName, "out.txt", "hello world\n" +
+                    "line in the middle\n" +
+                    "more text");
+
+            validateCapturedDocs(artifactName, "err.txt", "error line one\n" +
+                    "error line two");
+
+            validateCapturedDocs(artifactName, "out.matched.txt", "line in the middle\nmore text");
+            validateCapturedDocs(artifactName, "err.matched.txt", "error line one");
+            validateCapturedDocs(artifactName, "exitcode.txt", "5");
+        });
+    }
+
+    @Test
+    public void linesWithNotContain() {
+        nixOnly(() -> {
+            code(() -> {
+                cli.run("scripts/hello", ((output, error) -> {
+                    output.shouldNot(contain("line"));
+                }));
+            }).should(throwException(Pattern.compile("output\\[1]: equals \"line in the middle\"")));
+        });
+    }
+
+    private static void validateCapturedDocs(String artifactName, String fileName, String expectedContent) {
```

Comment:
This looks suspiciously similar to the stuff I did for http, is it worth extracting to somewhere common?

Reply:
I think my validation is overly simplified version of what you did. Maybe we will wait till a 3rd use case?

Reply:
Sounds good